### PR TITLE
core: add missing coverage

### PR DESCRIPTION
**Pull request by junr03**
Description: add areas of missing coverage across the core codebase.
Risk Level: low
Testing: adds more testing!

Signed-off-by: Jose Nino <jnino@lyft.com>

### CODE REVIEW

File: test/common/http/dispatcher_test.cc [hunk 1]
```diff
@@ -1324,5 +1324,127 @@ TEST_F(DispatcherTest, ResetWhenRemoteClosesBeforeLocal) {
...
+  Event::PostCb send_headers_post_cb;
+  EXPECT_CALL(event_dispatcher_, post(_)).WillOnce(SaveArg<0>(&send_headers_post_cb));
+  http_dispatcher_.sendHeaders(stream, c_headers, true);
+
+  EXPECT_CALL(request_decoder_, decodeHeaders_(_, true));
+  send_headers_post_cb();
+
+  // Encode 100 continue should blow up.
+  TestResponseHeaderMapImpl response_headers{{":status", "200"}};
+  EXPECT_DEATH(response_encoder_->encode100ContinueHeaders(response_headers), "panic: not implemented");
```
Comment (rebello95) [inline test/common/http/dispatcher_test.cc:1363]: Cool that this works

**Review by rebello95 — APPROVED** [review]

File: test/common/network/synthetic_address_impl_test.cc [hunk 2]
```diff
@@ -22,8 +22,10 @@ TEST(SyntheticAddressImplTest, Accessors) {
   SyntheticAddressImpl address;
   ASSERT_EQ(address.ip(), nullptr);
   ASSERT_EQ(address.pipe(), nullptr);
+  ASSERT_EQ(address.envoyInternalAddress(), nullptr);
   ASSERT_EQ(address.sockAddr(), nullptr);
   ASSERT_EQ(address.sockAddrLen(), 0);
+  address.socketInterface();
```
Comment (goaway) [inline test/common/network/synthetic_address_impl_test.cc:28]: ?

File: test/common/main_interface_test.cc [hunk 3]
```diff
@@ -218,6 +218,75 @@ TEST(MainInterfaceTest, ResetStream) {
...
+  terminate_engine(0);
+
+  ASSERT_TRUE(engine_cbs_context.on_exit.WaitForNotificationWithTimeout(absl::Seconds(10)));
+}
+
+TEST(MainInterfaceTest, InitEngineReturns1) {
+  // TODO(goaway): return new handle once multiple engine support is in place.
+  // https://github.com/lyft/envoy-mobile/issues/332
+  ASSERT_EQ(1, init_engine());
+}
```
Comment (goaway) [inline test/common/main_interface_test.cc:288]: This doesn't really feel like a useful test to me. I get that we're just trying to increase coverage, but I don't know that we actually want to be asserting or implying that this returns a specific value. If we were to change this to return 2 tomorrow, everything should still work, and the only thing that would be different is that now we'd have a broken test.

Comment (junr03) [reply test/common/main_interface_test.cc:288]: I see this as a sentinel that we have to have coverage for `init_engine()`. So in my mind having a test that breaks is, in and of itself, useful.

**Review by goaway — APPROVED** [review]

**Pull request merged**